logstash-core/lib/logstash/agent.rb

```diff
@@ -106,7 +106,7 @@ def stop_background_services
   def configure_metric
     if collect_metric?
       @logger.debug("Agent: Configuring metric collection")
-      @metric = LogStash::Instrument::Metric.create(:root)
```

Comment:
I rechecked the logic the namespace are created just outside the worker loop in the pipeline.
I understand the logic is a bit more harder to follow.

https://github.com/elastic/logstash/pull/4602/files#diff-bc1e9ea81caa54ea0d56abf4415fdf00R220

The problem is this part of the code actually update two different namespace and increment a counter. The pipeline specific and the application level counter for the number of `events_in`.

This is actually a local aggregation, to reduce complexity in the code base I preferred to update two different counter. What we could do in a future version is to allow to register aggregation callback or create aggregated view on metric store. I think this is a bit more complex to implement and it could be done in a future iteration.